qt.bzl

```diff
@@ -39,19 +39,15 @@ def qt_ui_library(name, ui, deps, **kwargs):
     )
 
 def _gencpp(ctx):
-    resource_files = [(f, ctx.actions.declare_file(f.path)) for f in ctx.files.files]
-    for target_file, output in resource_files:
-        ctx.actions.symlink(
-            output = output,
-            target_file = target_file,
-        )
+    """_gencpp runs the "resource compiler" (rcc) included with QT to encode a
+    set of input files as snippet of C++, which is saved to a file. Later rules
+    compile this C++ code and link it into binaries."""
 
-    args = ["--name", ctx.attr.resource_name, "--output", ctx.outputs.cpp.path, ctx.file.qrc.path]
-    ctx.actions.run(
-        inputs = [resource for _, resource in resource_files] + [ctx.file.qrc],
+    cmd = ["cat", ctx.file.qrc.path, "|", "rcc", "--name", ctx.attr.resource_name, "--output", ctx.outputs.cpp.path, "-"]
```

Comment:
I would have to try it out but I see two possible problems.
First I'm not sure that this will work on windows because relies on having cat installed. Maybe some select depending on the os could solve it.
Second, I'm not sure how good will play with toolchains, the idea was that rcc you can take it from the toolchain. I was intentionally trying to avoid run_shell because I though that in general should be prefered run over run_shell but I do not have enough knowledge to say why and I could not find in the documentation.
I will prepare a PR to set a Windows CI that would show the first case. Anyway for me it is fine to merge it and fix it after together with the Windows support, I just wanted to spot that can bring issues later.

Reply:
I saw that rules_go they also had a similar issue: https://github.com/bazelbuild/rules_go/issues/2463
How they solve it there was to use some prefix, this would be another option 
https://github.com/bazelbuild/rules_go/commit/695da5906684ab96e558c3159f7d88a20a6a1703

Reply:
Thanks for looking, those are both good points. It may be possible to change it to use `type` instead of `cat` under windows, but using a prefix like they do in rules_go sounds like a good approach. I'll take another look at this and see if I can get it working in a hopefully platform-agnostic way.